src/client/auth/gssapi.rs

```diff
@@ -0,0 +1,343 @@
+        let mut properties = GssapiProperties {
+            service_name: "mongodb".to_string(),
+            canonicalize_host_name: CanonicalizeHostName::None,
+            service_realm: None,
+            service_host: None,
+        };
+
+        if let Some(mechanism_properties) = &credential.mechanism_properties {
+            if let Some(Bson::String(name)) = mechanism_properties.get(SERVICE_NAME) {
+                properties.service_name = name.clone();
+            }
+
+            if let Some(canonicalize) = mechanism_properties.get(CANONICALIZE_HOST_NAME) {
+                properties.canonicalize_host_name = match canonicalize {
+                    Bson::String(s) => match s.as_str() {
+                        "none" => CanonicalizeHostName::None,
+                        "forward" => CanonicalizeHostName::Forward,
+                        "forwardAndReverse" => CanonicalizeHostName::ForwardAndReverse,
+                        _ => {
+                            return Err(Error::authentication_error(
+                                GSSAPI_STR,
+                                format!(
+                                    "Invalid CANONICALIZE_HOST_NAME value: {}. Valid values are \
+                                     'none', 'forward', 'forwardAndReverse'",
+                                    s
+                                )
+                                .as_str(),
+                            ))
+                        }
+                    },
+                    Bson::Boolean(true) => CanonicalizeHostName::ForwardAndReverse,
+                    Bson::Boolean(false) => CanonicalizeHostName::None,
+                    _ => {
+                        return Err(Error::authentication_error(
+                            GSSAPI_STR,
+                            "CANONICALIZE_HOST_NAME must be a string or boolean",
+                        ))
+                    }
+                };
+            }
+
+            if let Some(Bson::String(realm)) = mechanism_properties.get(SERVICE_REALM) {
+                properties.service_realm = Some(realm.clone());
+            }
+
+            if let Some(Bson::String(host)) = mechanism_properties.get(SERVICE_HOST) {
+                properties.service_host = Some(host.clone());
+            }
+        }
+
+        Ok(properties)
+    }
+}
+
+struct GssapiAuthenticator {
+    pending_ctx: Option<PendingClientCtx>,
+    established_ctx: Option<ClientCtx>,
+    user_principal: Option<String>,
+    is_complete: bool,
+}
+
+impl GssapiAuthenticator {
+    // Initialize the GssapiAuthenticator by creating a PendingClientCtx and
+    // getting an initial token to send to the server.
+    async fn init(
+        user_principal: Option<String>,
+        properties: GssapiProperties,
+        hostname: &str,
+    ) -> Result<(Self, Vec<u8>)> {
+        let service_name: &str = properties.service_name.as_ref();
+        let mut service_principal = format!("{}/{}", service_name, hostname);
```

Comment:
```suggestion
        let mut service_principal = format!("{service_name}/{hostname}");
```
Let's just fix this before clippy starts whining about it the next time they update it :)